core/src/main/java/org/opensearch/sql/calcite/utils/OpenSearchTypeFactory.java

```diff
@@ -151,6 +152,13 @@ public static ExprType convertRelDataTypeToExprType(RelDataType type) {
         return TIMESTAMP;
       case GEOMETRY:
         return IP;
+      case INTERVAL_YEAR:
+      case INTERVAL_MONTH:
+      case INTERVAL_DAY:
+      case INTERVAL_HOUR:
+      case INTERVAL_MINUTE:
+      case INTERVAL_SECOND:
```

Comment:
spark-ppl support YearMonthIntervalType and DayTimeIntervalType? if yes, we can track it as issue and fix later.

Reply:
no, YearMonthIntervalType and DayTimeIntervalType are not supported in spark-ppl
code code is not in use for now.